plugins/out_http/http.c

```diff
@@ -276,7 +276,15 @@ static int http_post(struct flb_out_http *ctx,
                 flb_plg_error(ctx->ins, "%s:%i, HTTP status=%i",
                               ctx->host, ctx->port, c->resp.status);
             }
-            out_ret = FLB_RETRY;
+            if (c->resp.status >= 400 && c->resp.status < 500 && c->resp.status != 429) {
+                flb_plg_warn(ctx->ins, "could not flush records to %s:%i (http_do=%i), "
+                                "chunk will not be retried",
+                                ctx->host, ctx->port, ret);
+                out_ret = FLB_ERROR;
```

Comment:
Does that mean logs will be lost when the endpoint is unavailable?

Reply:
if return a code in 4xx range except 429, this means that the are an error in client side in consequence that records will not be retried.

Reply:
Fluentd has a setting for retryable response codes: https://docs.fluentd.org/output/http#retryable_response_codes maybe we could add this as well?